File: Selenium/Activities/Activity12.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity12 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get(" https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of webpage:" + driver.getTitle());

        WebElement textField=driver.findElement(By.id("input-text"));
        boolean checkIfEnabled=textField.isEnabled();
        System.out.println("Text field enabled="+checkIfEnabled);

        driver.findElement(By.id("toggleInput")).click();
        checkIfEnabled=textField.isEnabled();
        System.out.println("Text field enabled="+checkIfEnabled);
        driver.close();
    }
}

File: Selenium/Activities/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    wait = WebDriverWait(driver, 10)
    print("Title of webpage:" + driver.title)
    textField = driver.find_element(By.ID, "input-text")
    checkIfEnabled = textField.is_enabled()
    print("Text field enabled=" + str(checkIfEnabled))
    driver.find_element(By.ID, "toggleInput").click()
    checkIfEnabled = textField.is_enabled()
    print("Text field enabled=" + str(checkIfEnabled))
    driver.close()
